audformat/core/utils.py

```diff
@@ -1074,3 +1074,24 @@ def union(
     index = index.drop_duplicates()
 
     return index
+
+
+def iter_index_by_file(
+        index: pd.Index
+) -> (str, typing.Iterator[pd.Index]):
+    r"""Iterate over index by file.
+
+    In each iteration return sub-index for one file.
+
+    Args:
+        index: index object conform to
+            :ref:`table specifications <data-tables:Tables>`
+
+    Returns:
+        iterator in form of (file, sub_index)
```

Comment:
Could you please add an docstring example for this.
That might then even cover the code already.

Reply:
Or in code https://github.com/audeering/audformat/blob/eeb55de67a1d6fd06e448117f40e8ecda9cbc029/audformat/core/utils.py#L258-L267

Reply:
I've added the docstring example, and the code is covered as you said

Reply:
Thanks.